### PR TITLE
Discord Bot Disconnect

### PR DESCRIPTION
**Pull request by timolegros**
## Link to Issue
Closes: #5853 
Closes: #7146

## Description of Changes
- Adds `/removeDiscordBotConfig` route
- Switch button from 'Reconnect' to 'Disconnect'
  - 'Reconnect' just retried the flow for connecting which actually throws an error and leaves the config in an inconsistent state. It also makes it impossible to switch which Discord server the Discord bot is connected to.
- Added `discordConfigId` to redirect URL to ensure integration is reflected on management page if successful (means a refresh is not required to view the Discord connection).

## Test Plan
- Navigate to the manage integrations page of any community and connect + disconnect discord bot
- Link the bot to a topic and post on Discord -> should see posts appear on commonwealth

## Deployment Plan

1. 

## Other Considerations

-

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/pages/CommunityManagement/Integrations/Discord/Discord.tsx [hunk 1]
```diff
@@ -22,6 +25,13 @@ const Discord = () => {
   const [connectionStatus, setConnectionStatus] = useState<ConnectionStatus>(
     community.discordConfigId ? 'connected' : 'none',
   );
+
+  const queryClient = useQueryClient();
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/CommunityManagement/Integrations/Discord/Discord.tsx:29]: Do not pollute component with this. See my other comment.

File: packages/commonwealth/client/scripts/views/pages/CommunityManagement/Integrations/Discord/Discord.tsx [hunk 2]
```diff
@@ -22,6 +25,13 @@ const Discord = () => {
...
+
+  const queryParams = new URLSearchParams(window.location.search);
+  if (queryParams.has('discordConfigId')) {
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/CommunityManagement/Integrations/Discord/Discord.tsx:32]: fix the warnings

File: packages/commonwealth/client/scripts/views/pages/CommunityManagement/Integrations/Discord/Discord.tsx [hunk 3]
```diff
@@ -57,16 +68,45 @@ const Discord = () => {
...
+      return;
+    try {
+      await axios.post(`${app.serverUrl()}/removeDiscordBotConfig`, {
+        community_id: app.activeChainId(),
+        jwt: app.user.jwt,
+      });
+      // remove channel query from cache to force refresh
+      queryClient.removeQueries([ApiEndpoints.DISCORD_CHANNELS], {
+        exact: true,
+      });
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/CommunityManagement/Integrations/Discord/Discord.tsx:95]: We can't do this anymore. We are in the process of tidy up the mess so please align with new pattern.
Basically API calls should by done with react-query and all of these api related logic should be handled in `state/api` directory, following existing patterns. As @mzparacha worked recently in this area, let's make sure that it is done properly

**Review by masvelio — CHANGES_REQUESTED** [review]
move API call logic to react-query in `state/api`

File: packages/commonwealth/client/scripts/state/api/discord/removeDiscordBotConfig.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import { ApiEndpoints, queryClient } from 'state/api/config';
+
+interface RemoveDiscordBotConfigProps {
+  communityId: string;
+}
+
+const removeDiscordBotConfig = async ({
+  communityId,
+}: RemoveDiscordBotConfigProps) => {
+  await axios.post(`${app.serverUrl()}/removeDiscordBotConfig`, {
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/state/api/discord/removeDiscordBotConfig.ts:13]: lets move `/removeDiscordBotConfig` as const to `/client/scripts/state/api/config.ts`

**Review by mzparacha — APPROVED** [review]

**Pull request merged**